python/pyspark/pandas/series.py

```diff
@@ -6859,13 +6860,16 @@ def _reduce_for_stat_function(
         sfun : the stats function to be used for aggregation
         name : original pandas API name.
         axis : used only for sanity check because series only support index axis.
-        numeric_only : not used by this implementation, but passed down by stats functions
+        numeric_only : not used by this implementation, but passed down by stats functions.
         """
         axis = validate_axis(axis)
         if axis == 1:
             raise NotImplementedError("Series does not support columns axis.")
 
-        scol = sfun(self)
+        if skipna:
+            scol = sfun(self)
+        else:
+            scol = F.first(F.lit(np.nan))
```

Comment:
Hm,  shouldn't we apply the given function `sfun`?

Reply:
Good catch! Fixed. Test cases when Series has no NAs and `skipna` is False are added as well.